language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/workspace/BallerinaWorkspaceManager.java

```diff
@@ -296,20 +296,22 @@ public Optional<SemanticModel> semanticModel(Path filePath, @Nonnull CancelCheck
      * Returns module compilation from the file path provided.
      *
      * @param filePath file path of the document
+     * @param isSourceChange True if the given file's source is changed
      * @return {@link ModuleCompilation}
      */
     @Override
-    public Optional<PackageCompilation> waitAndGetPackageCompilation(Path filePath) {
+    public Optional<PackageCompilation> waitAndGetPackageCompilation(Path filePath, boolean isSourceChange) {
```

Comment:
```suggestion
    public Optional<PackageCompilation> waitAndGetPackageCompilation(Path filePath, LSOperation operation) {
```

Shall we use the https://github.com/ballerina-platform/ballerina-lang/blob/9ead5e64f7944ca75f93b60ca8abef3b77970627/language-server/modules/langserver-commons/src/main/java/org/ballerinalang/langserver/commons/LSOperation.java#L25 instead and check if the operation is a DidChangeOperation

Reply:
https://github.com/ballerina-platform/ballerina-lang/pull/39190/files#r1064370919

Reply:
https://github.com/ballerina-platform/ballerina-lang/pull/39190#discussion_r1065731649